x-pack/plugins/lens/public/app_plugin/lens_top_nav.tsx

```diff
@@ -411,13 +411,24 @@ export const LensTopNavMenu = ({
         const dataViewId = datasourceMap[activeDatasourceId].getUsedDataView(
           datasourceStates[activeDatasourceId].state
         );
-        const dataView = await data.dataViews.get(dataViewId);
-        setCurrentIndexPattern(dataView ?? indexPatterns[0]);
+        try {
+          const dataView = dataViewId ? await data.dataViews.get(dataViewId) : undefined;
+          setCurrentIndexPattern(dataView ?? indexPatterns[0]);
+        } catch (e) {
+          // silent fail
```

Comment:
Why this change?

Reply:
Not needed here, I removed the try catch, was a leftover from a previous implementation I was testing